R/RLearner_classif_mxff.R

```diff
@@ -0,0 +1,416 @@
+#' @export
+makeRLearner.classif.mxff = function() {
+  makeRLearnerClassif(
+    cl = "classif.mxff",
+    package = "mxnet",
+    par.set = makeParamSet(
+      # architectural hyperparameters
+      makeIntegerLearnerParam(id = "layers", lower = 1L, upper = 3L, default = 1L),
+      makeIntegerLearnerParam(id = "num.layer1", lower = 1L, default = 1L),
+      makeIntegerLearnerParam(id = "num.layer2", lower = 1L, default = 1L,
+        requires = quote(layers > 1)),
+      makeIntegerLearnerParam(id = "num.layer3", lower = 1L, default = 1L,
+        requires = quote(layers > 2)),
+      makeDiscreteLearnerParam(id = "act1", default = "tanh",
+        values = c("tanh", "relu", "sigmoid", "softrelu")),
+      makeDiscreteLearnerParam(id = "act2", default = "tanh",
+        values = c("tanh", "relu", "sigmoid", "softrelu"),
+        requires = quote(layers > 1)),
+      makeDiscreteLearnerParam(id = "act3", default = "tanh",
+        values = c("tanh", "relu", "sigmoid", "softrelu"),
+        requires = quote(layers > 2)),
+      makeDiscreteLearnerParam(id = "act.out", default = "softmax",
+        values = c("rmse", "softmax", "logistic")),
+      makeLogicalLearnerParam(id = "conv.layer1", default = FALSE),
+      makeLogicalLearnerParam(id = "conv.layer2", default = FALSE,
+        requires = quote(layers > 1 && conv.layer1 == TRUE)),
+      makeLogicalLearnerParam(id = "conv.layer3", default = FALSE,
+        requires = quote(layers > 2 && conv.layer2 == TRUE)),
+      makeIntegerVectorLearnerParam(id = "conv.data.shape",
+        requires = quote(conv.layer1 == TRUE)),
+      makeIntegerLearnerParam(id = "conv.kernel11", lower = 1L,
+        requires = quote(conv.layer1 == TRUE)),
+      makeIntegerLearnerParam(id = "conv.kernel12", lower = 1L,
+        requires = quote(conv.layer1 == TRUE)),
```

Comment:
I think in principle you're right and that's the implementation I had originally. However, there is a problem with `integerVectorParams` when using them with `generateDesign` because it the vector parameters get split up into single integer parameters which can in turn not be used as input for the learner. We talked about this issue when it occurred :)
```
> test.ps = makeParamSet(makeIntegerVectorParam(id = "test", len = 2, lower = 0, upper = 10))
> generateDesign(n = 2, test.ps)

   test1 test2
1     7     2
2     2     8
```